optimum/habana/transformers/models/mpt/modeling_mpt.py

```diff
@@ -90,13 +101,32 @@ def gaudi_mpt_attention_forward(
     if attention_mask is not None:
         attention_scores = attention_scores.masked_fill(attention_mask, torch.finfo(query_states.dtype).min)
 
-    # (batch_size, n_heads, seq_length, key_length)
-    attn_weights = nn.functional.softmax(attention_scores.float(), dim=-1).to(value_states.dtype)
-    attn_weights = nn.functional.dropout(attn_weights, p=self.attn_dropout_p, training=self.training)
+    if use_flash_attention and FusedSDPA:
+        import habana_frameworks.torch.hpu as ht
+
+        with ht.sdp_kernel(enable_recompute=flash_attention_recompute):
+            attn_output = FusedSDPA.apply(
```

Comment:
fusedSDPA should replace the whole attention block when --use_flash_attention is enabled.
but this commit inserted fusedSDPA in the middle of attention after q*k.
you should be very careful when you make changes in the model and run thorough tests

Reply:
updated